skyportal/handlers/api/earthquake.py

```diff
@@ -0,0 +1,616 @@
+
+    event = session.scalars(
+        EarthquakeEvent.select(user).where(EarthquakeEvent.event_id == event_id)
+    ).first()
+
+    if event is None:
+        event = EarthquakeEvent(
+            event_id=event_id, event_uri=event_uri, sent_by_id=user.id
+        )
+        session.add(event)
+    else:
+        if not event.is_accessible_by(user, mode="update"):
+            raise ValueError(
+                "Insufficient permissions: Earthquake event can only be updated by original poster"
+            )
+
+    earthquake_notice = EarthquakeNotice(
+        content=payload.encode('utf-8'),
+        event_id=event_id,
+        lat=origin.latitude,
+        lon=origin.longitude,
+        depth=origin.depth,
+        magnitude=mag,
+        date=origin.time.datetime,
+        sent_by_id=user.id,
+    )
+    session.add(earthquake_notice)
+    session.commit()
+
+    return event.id
+
+
+def post_earthquake_from_dictionary(payload, user_id, session):
+    """Post Earthquake to database from dictionary.
+    payload: dict
+        Dictionary containing date and skymap
+    user_id : int
+        SkyPortal ID of User posting the Earthquake
+    session: sqlalchemy.Session
+        Database session for this transaction
+    """
+
+    user = session.query(User).get(user_id)
+
+    date = payload['date']
+    event_id = payload['event_id']
+    latitude = payload['latitude']
+    longitude = payload['longitude']
+    depth = payload['depth']
+    magnitude = payload['magnitude']
+
+    event = session.scalars(
+        EarthquakeEvent.select(user).where(EarthquakeEvent.event_id == event_id)
+    ).first()
+
+    if event is None:
+        event = EarthquakeEvent(event_id=event_id, sent_by_id=user.id)
+        session.add(event)
+    else:
+        if not event.is_accessible_by(user, mode="update"):
+            raise ValueError(
+                "Insufficient permissions: Earthquake event can only be updated by original poster"
+            )
+
+    earthquake_notice = EarthquakeNotice(
+        event_id=event_id,
+        lat=latitude,
+        lon=longitude,
+        depth=depth,
+        magnitude=magnitude,
+        date=date,
+        sent_by_id=user.id,
+    )
+    session.add(earthquake_notice)
+    session.commit()
+
+    return event.id
+
+
+class EarthquakeStatusHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        """
+        ---
+        description: Get all Earthquake status tags
+        tags:
+          - earthquakeevents
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        with self.Session() as session:
+            statuses = (
+                session.scalars(sa.select(EarthquakeEvent.status).distinct())
+                .unique()
+                .all()
+            )
+            return self.success(data=statuses)
```

Comment:
This will return everything ever posted. What is this used for? Just to get a list of possible statuses?

Reply:
yeah will be used for filtering on the frontend.